packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_sidebar.dart

```diff
@@ -0,0 +1,277 @@
+    return Column(
+      children: [
+        itemDivider,
+        for (final property in _properties) ...[
+          _EditablePropertyItem(property: property),
+          itemDivider,
+        ],
+      ],
+    );
+  }
+}
+
+class _EditablePropertyItem extends StatelessWidget {
+  const _EditablePropertyItem({required this.property});
+
+  final _WidgetProperty property;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Flexible(
+          flex: 3,
+          child: Padding(
+            padding: const EdgeInsets.all(_PropertiesList.itemPadding),
+            child: _PropertyInput(property: property),
+          ),
+        ),
+        if (property.isRequired || property.isDefault) ...[
+          Flexible(child: _PropertyLabels(property: property)),
+        ] else
+          const Spacer(),
+      ],
+    );
+  }
+}
+
+class _PropertyLabels extends StatelessWidget {
+  const _PropertyLabels({required this.property});
+
+  final _WidgetProperty property;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final isRequired = property.isRequired;
+    final isDefault = property.isDefault;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        if (isRequired)
+          Padding(
+            padding: const EdgeInsets.all(_PropertiesList.itemPadding),
+            child: RoundedLabel(
+              labelText: 'required',
+              backgroundColor: colorScheme.primary,
+              textColor: colorScheme.onPrimary,
+            ),
+          ),
+        if (isDefault)
+          const Padding(
+            padding: EdgeInsets.all(_PropertiesList.itemPadding),
+            child: RoundedLabel(labelText: 'default'),
+          ),
+      ],
+    );
+  }
+}
+
+class _PropertyInput extends StatelessWidget {
+  const _PropertyInput({required this.property});
+
+  final _WidgetProperty property;
+
+  @override
+  Widget build(BuildContext context) {
+    final decoration = InputDecoration(
+      helperText: '',
+      errorText: property.errorText,
+      isDense: true,
+      label: Text(property.name),
+      border: const OutlineInputBorder(),
+    );
+
+    switch (property.type) {
+      case 'enum':
+      case 'bool':
+        final options =
+            property.type == 'bool' ? ['true', 'false'] : property.options;
+        return DropdownButtonFormField(
+          value: property.valueDisplay,
+          decoration: decoration,
+          items:
+              (options ?? []).map((option) {
+                return DropdownMenuItem(
+                  value: option,
+                  onTap: () {},
+                  child: Text(option),
+                );
+              }).toList(),
+          onChanged: (_) {},
+        );
+      case 'double':
+      case 'int':
+      case 'string':
+        return TextFormField(
```

Comment:
consider using `DevToolsClearableTextField` so that we get the default devtools styling and a handy clear action